### PR TITLE
[FEATURE] PMP-1841 popup

### PR DESCRIPTION
**Pull request by bsparks**
setup a capability api for configuring the toolbar and drag per part type; start of the feature that will enable popup and others to configure more about what toolbar capabilities are available.

### CODE REVIEW

File: assets/src/components/activities/adaptive/AdaptiveAuthoring.tsx [hunk 1]
```diff
@@ -68,19 +72,31 @@ const Adaptive = (
...
     if (selectedPartId) {
       const part = parts.find((p) => p.id === selectedPartId);
       if (part) {
-        setSelectedPart(part);
+        let capabilities = { ...defaultCapabilities };
+        // attempt to get an instance of the part class
+        const PartClass = customElements.get(part.type);
+        if (PartClass) {
+          const instance = new PartClass();
+          if (instance.getCapabilities) {
```
Comment (dsuriano) [inline assets/src/components/activities/adaptive/AdaptiveAuthoring.tsx:85]: image

File: assets/src/components/parts/types/parts.ts [hunk 2]
```diff
@@ -70,3 +70,13 @@ export interface ColorPalette {
   lineStyle?: number;
   lineThickness?: number;
 }
+
+export const defaultCapabilities = {
+  move: true,
```
Comment (dsuriano) [inline assets/src/components/parts/types/parts.ts:75]: will `move` also encompass `z` as well as `x` & `y` ?

Comment (bsparks) [reply assets/src/components/parts/types/parts.ts:75]: yes, I was thinking so, and have enabled both drag and the fwd/back buttons based on it

File: assets/src/components/activities/adaptive/AdaptiveAuthoring.tsx (hunk 1 above)
Comment (bsparks) [reply assets/src/components/activities/adaptive/AdaptiveAuthoring.tsx:85]: hmm it didn't complain for me, nor did github...

Comment (dsuriano) [reply assets/src/components/activities/adaptive/AdaptiveAuthoring.tsx:85]: yeah, i didn't see any compilation errors and all the tests passed... so 🤷‍♂️

**Review by dsuriano — APPROVED** [review]

**Pull request merged**